### ISSUE TITLE
Incorrect global phase added to qobj header

### ISSUE THREAD

**Issue by chriseclectic** [Qiskit/qiskit-terra#5125]
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.7
- **Operating system**: MacOS

### What is the current behavior?

I think #4915 introduced some bugs in how the the global phase header is calculated which is causing some Aer tests to fail. It appears certain unitary gate instructions are causing a non-zero phase to be added during assemble, even though the unitary matrix is being added as an instruction without any modification. So far I've confirmed this happens for several single qubit gate matrices:

### Steps to reproduce the problem

The following code checks the original matrix is inserted as the qobj instruction parameters, and what global phase is added to the circuit. The global phase should be zero in all cases

```python
from qiskit import QuantumCircuit, assemble
from qiskit.quantum_info import Operator
from qiskit.circuit.library import *

for gate in IGate, XGate, YGate, ZGate, HGate, SGate, SdgGate, TGate, TdgGate, SXGate:
    U = qi.Operator(gate())
    qc = QuantumCircuit(1)
    qc.append(U, [0])
    qobj = assemble(qc)
    global_phase = qobj.experiments[0].header.global_phase
    params = qobj.experiments[0].instructions[0].params
    
    print('{}, params == U: {}, global_phase: {}'.format(
        gate.__name__, U == qi.Operator(params[0]), global_phase))
```
prints

```
IGate, params == U: True, global_phase: 0.0
XGate, params == U: True, global_phase: -1.5707963267948966
YGate, params == U: True, global_phase: 1.5707963267948966
ZGate, params == U: True, global_phase: 0.0
HGate, params == U: True, global_phase: 0.0
SGate, params == U: True, global_phase: -2.220446049250313e-16
SdgGate, params == U: True, global_phase: 2.220446049250313e-16
TGate, params == U: True, global_phase: 5.551115123125783e-17
TdgGate, params == U: True, global_phase: -5.551115123125783e-17
SXGate, params == U: True, global_phase: 0.7853981633974482
```
### What is the expected behavior?

Adding unitary matrix gates should not change the global phase of the circuit during assembly.

**Comment by mtreinish** [comment]
Is this just an issue with Qobj assembly or is global phase incorrect in the circuit attribute too?

**Comment by chriseclectic** [comment]
This just appears to be due to `assemble`. It also applies to standard (non-unitary) gates with phases when they are unrolled. For example:

```
theta = 0.5
qc = QuantumCircuit(1)
qc.rz(theta, 0)
qobj = assemble(qc)
global_phase = qobj.experiments[0].header.global_phase
```

Will have a qobj with an "rz" instruction in it and a global phase of -theta/2. But the global phase should only be added if rz is unrolled to u1, if the qobj has the "rz" instruction the global phase should be 0

**Comment by kdk** [comment]
Resolved by #5126 .
